### PR TITLE
Embed software metadata, publication and citation

### PR DESCRIPTION
**Pull request by sdruskat**
As [announced earlier](https://github.com/CaptainSifff/paper_teaching-learning-RSE/pull/214#issuecomment-1853435812), this PR provides additions to @ljgarcia's PR on ML and metadata (#214).

Let me know if I should put this up as a dedicated PR to `main` once #214 is merged.

I'd be happy about feedback regarding these changes, and will try to join today's call.

### CODE REVIEW

**Comment by mhagdorn** [comment]
Thanks, these are great additions.

**Review by MakisH — APPROVED** [review]

File: competencies.md [hunk 1]
```diff
@@ -471,16 +473,22 @@ existing work instead. To (re-) use software, researchers have to be able to
...
 and making them available for re-use by the research community and the general public.
 RSEs need to have a basic understanding of common software licence types, such as "proprietary", "copyleft", and "permissive",
 the compatibility of different common licences and the ramifications for re-using and composing programs.
-Finally, RSEs will need to properly execute the technicalities of software publishing,
+Beyond that, RSEs will need to properly execute the technicalities of software publishing,
 such as applying licences, honouring copyright statements and crediting contributors.
+RSEs also need to understand and apply principles and good practices of software publication.
+This includes the collection, curation and provision of relevant software publication metadata [@druskat_hermes_2022]
+both for software citation [@druskat_cff_2021], and
+for generic and domain-specific software description [@jones_codemeta_2017].
```
Comment (CaptainSifff) [inline competencies.md:491]: So far, our competencies have been kind of intentionally without a lot of references(which can be critisized), and now we have three here, which (I think) highlight today's technical solutions. Do we have a *single, good* reference, that highlight's the importance of metadata for science on the same abstract level as we have here? Then I would move the technical implementations to the current-day-contextualization(Where some of them already are).

**Review by CaptainSifff — COMMENTED** [review]
I think we should discuss the amount of refs for the general competencies. Currently they are written, to stand for themselves, without requiring the reader to delve into the literature.

File: competencies.md (hunk 1 above)
Comment (sdruskat) [reply competencies.md:491]: Thanks @CaptainSifff, I'll have a look and adapt.

**Comment by BeastyBlacksmith** [comment]
Sorry for the unintended close. You could target main with this PR now

File: competencies.md (hunk 1 above)
Comment (sdruskat) [reply competencies.md]: Done in cf67856871466a8cdc24290cdc81492fba2ba033.

**Comment by sdruskat** [comment]
@mhagdorn @CaptainSifff I've now removed the extraneous references in the competency overview and left only one that talk more generally about software metadata. In lieu of a canonical reference paper, this is the most comprehensive overview at the moment I think (although disclaimer: I'm an author). I can still remove it if you feel it shouldn't be there, and no hard feelings ;).

**Review by CaptainSifff — APPROVED** [review]
Great, thanks!

**Pull request merged**